From 9f21be00463e8bc13137f33f5d970789c2047511 Mon Sep 17 00:00:00 2001
From: Md Sadre Alam <quic_mdalam@quicinc.com>
Date: Wed, 1 Nov 2023 11:47:14 +0530
Subject: [PATCH] crypto: qce - Add support for command descriptor flag

Add support for command descriptor flag. This flag will
be set by device tree or by dt parsing from boot args.
based on this flag only crypto register read/write will
be done via bam dma.

Change-Id: Ifd8b3cac38b16d02ee0c06a00a8c4daead9fed60
Signed-off-by: Md Sadre Alam <quic_mdalam@quicinc.com>
---
 drivers/crypto/qce/core.c | 3 +++
 drivers/crypto/qce/core.h | 1 +
 2 files changed, 4 insertions(+)

diff --git a/drivers/crypto/qce/core.c b/drivers/crypto/qce/core.c
index 3b6be6d91b17..26577f06c729 100644
--- a/drivers/crypto/qce/core.c
+++ b/drivers/crypto/qce/core.c
@@ -524,6 +524,9 @@ static int qce_crypto_probe(struct platform_device *pdev)
 	if (ret < 0)
 		return ret;
 
+	if (device_property_read_bool(dev, "qce,cmd_desc_support"))
+		qce->qce_cmd_desc_enable = true;
+
 	if (device_property_read_bool(dev, "qce,use_fixed_hw_key"))
 		qce->use_fixed_key = true;
 
diff --git a/drivers/crypto/qce/core.h b/drivers/crypto/qce/core.h
index 81ec5d40f49e..c6284edd209d 100644
--- a/drivers/crypto/qce/core.h
+++ b/drivers/crypto/qce/core.h
@@ -95,6 +95,7 @@ struct qce_device {
 	bool use_fixed_key;
 	struct kobject kobj;
 	struct kobject *kobj_parent;
+	bool qce_cmd_desc_enable;
 };
 
 /**
-- 
2.34.1

